.github/workflows/ci.yaml

```diff
@@ -57,6 +57,111 @@ jobs:
+            exe: nextstrain
+
+          - os: windows-2019
+            target: x86_64-pc-windows-msvc
+            exe: nextstrain.exe
+
+    runs-on: ${{ matrix.os }}
+    defaults:
+      run:
+        shell: bash
+    steps:
+      # Install pyoxidizer.
+      #
+      # Even though it's a Rust project, the easiest cross-platform way to
+      # install it is via pip since they publish wheels with the binaries. :-)
+      # Note that this Python version doesn't impact the actual build.
+      - uses: actions/setup-python@v3
+        with:
+          python-version: "3.10"
+      - run: pip install pyoxidizer
+
+      # Build the executable + necessary external files from the dists.
+      - uses: actions/checkout@v3
+      - uses: actions/download-artifact@v3
+        with:
+          name: dist
+          path: dist/
+
+      - run: |
+          pyoxidizer build \
+            --release \
+            --target-triple ${{ matrix.target }} \
+            --var NEXTSTRAIN_CLI_DIST dist/nextstrain_cli-*-py3-none-any.whl
+
+      # Analyze the executable for potential portability issues.
+      #
+      # This is for informational purposes only in build logs, so we don't care
+      # if it fails.  Currently it only works on Linux, though it's supposed to
+      # eventually work on all platforms supported by pyoxidizer.
+      - run: pyoxidizer analyze build/${{ matrix.target }}/release/installation/${{ matrix.exe }}
+        continue-on-error: true
+        if: runner.os == 'Linux'
+
+      # XXX TODO: Review and report on licensing of all the stuff built into
+      # the binary, as bundling things statically can trigger different license
+      # terms than "normal" installs (e.g. via pip).  See also pyoxidizer's
+      # docs about this and the tooling it includes to support license review.¹
+      #   -trs, 1 June 2022
+      #
+      # ¹ https://pyoxidizer.readthedocs.io/en/stable/pyoxidizer_packaging_licensing.html#licensing-considerations
+
+      # Create installation archive.
+      #
+      # Use tar on Unix to preserve file modes (e.g. the executable bit), thus
+      # avoiding having to restore them manually after archive extraction.  Use
+      # zip on Windows because it's a native format which requires no extra
+      # tooling.
+      - run: tar czvpf ${{ matrix.target }}.tar.gz -C build/${{ matrix.target }}/release/installation/ .
+        if: runner.os != 'Windows'
+
+      - run: Compress-Archive -DestinationPath ${{ matrix.target }}.zip -Path build/${{ matrix.target }}/release/installation/*
+        shell: pwsh
+        if: runner.os == 'Windows'
+
+      # Upload installation archive as a workflow artifact.
+      #
+      # At least one path needs to match, or this errors.
+      - uses: actions/upload-artifact@v3
+        with:
+          name: ${{ matrix.target }}
+          path: |
+            ${{ matrix.target }}.tar.gz
+            ${{ matrix.target }}.zip
+          if-no-files-found: error
```

Comment:
Nod. There's no way to avoid nested compression with GitHub Action's artifacts. They're documented to be always wrapped up in a zip container on download.

But it doesn't matter, because for actually using these standalone builds we'd publish them as assets on a GitHub Release (or assets in some S3 bucket somewhere), not directly via GitHub Actions artifacts (which besides being awkward to use also require authn to download anyway).